solutions/banking/watson-scripts/assistant-project-creation.sh

```diff
@@ -7,7 +7,19 @@ WATSON_ASSISTANT_URL=$2
 
 token="$(echo "$IAM_TOKEN" | awk '{print $2}')"
 
-curl -X POST --location "$WATSON_ASSISTANT_URL/v2/assistants?version=2023-06-15" \
+ASSISTANT_ID=$(curl -X GET --location "$WATSON_ASSISTANT_URL/v2/assistants?version=2023-06-15" \
     --header "Authorization: Bearer $token" \
     --header "Content-Type: application/json" \
-    --data "{\"name\":\"gen-ai-rag-sample-app-assistant\",\"language\":\"en\",\"description\":\"Generative AI sample app assistant\"}"
+    | jq -r '.assistants[] | select(.name == "gen-ai-rag-sample-app-assistant") | .assistant_id ')
+
+if [[ -z "$ASSISTANT_ID" ]]; then
+  # if not ASSISTANT_ID is found then create a new assistant project
+  curl -X POST --location "$WATSON_ASSISTANT_URL/v2/assistants?version=2023-06-15" \
```

Comment:
will this fail the script if the curl fails? I don't think it will. I think you need to add one of the `--fail` flags so the command returns a non zero output and will fail the script. See https://curl.se/docs/manpage.html